MapboxNavigation/NavigationMapView.swift

```diff
@@ -427,36 +426,96 @@ open class NavigationMapView: MGLMapView {
             }
         }
     }
-    
     /**
-     Fired when the user taps a route. If there is an alternate route, and an alternate route is tapped, it is selected.
+     Fired when NavigationMapView detects a tap not handled elsewhere by other gesture recognizers.
      */
-    func didTapRoute(tap: UITapGestureRecognizer) {
+    func didRecieveTap(sender: UITapGestureRecognizer) {
         guard let routes = routes else { return }
         
-        let tapPoint = tap.location(in: self)
-        let tapCoordinate = self.convert(tapPoint, toCoordinateFrom: tap.view)
+        let waypointTest = waypoints(on: routes, closeTo: sender) //are there waypoints near the tapped location?
+        if let activeRoute = waypointTest?.multipointRoutes.first ,let waypointToRemove = waypointTest?.waypoints.first { //test passes
+            return requestNew(route: activeRoute, without: waypointToRemove)
+        } else if let routes = self.routes(closeTo: sender) {
+            guard let selectedRoute = routes.first else { return }
+            select(route: selectedRoute)
```

Comment:
This logic should be moved to the delegate implementation. We can make the helper methods public to facilitate that implementation. After all, the developer may want to programmatically select a route, or they may want to determine the route closest to a view that’s floating above the map view.

If implementing this logic represents an undue burden on the developer, we should vend a prefabricated view controller (#808). Alternatively, we could explore supplementing the existing Objective-C delegate protocol with a Swift protocol that can come with its own default implementations.